proposals/20210525-exmplefilter.md

```diff
@@ -0,0 +1,70 @@
+#### SIG TFX-Addons
+# Project Proposal
+## Instructions
+Copy this template, naming it with the date in `yyyymmdd` format followed by a short name, and fill in the blanks below.
+When you're ready for review, place it in `/proposals` and create a pull request.
+
+---
+
+**Your name:** Ryan Clough
+
+**Your email:** rclough@spotify.com
+
+**Your company/organization:** Spotify
+
+**Project name:** Example Filter
+
+## Project Description
+Beam bassed component that can filter Examples based on a user-defined predicate function.
+
+## Project Category
+Choose 1: Component
+
+## Project Use-Case(s)
+Data can be imported into TFX in a number of ways, and indeed, sometimes the dataset you wish to load is not under your direct 
+control. In cases like these, it is useful to have a component that can filter your input data with simple rules. Ex: filter 
+all records where `feature_a >= 1`. 
+
+Our organization currently has a component for this purpose that is in active use. It is not ass robust as it could be.
+
+It is also worth conidering that we may wish to try and promote this functionality to be included in the TFX core base ExampleGen,
+so that the filtering could be done within any ExampleGen based component.
+
+## Project Implementation
+Spotify can provide the current implementation, which is based off of an old version of Tensorflow Transform. At a high level, use
+of the coomponent looks like:
+
+```python
+def perdicate_fn(example)
+    # Throw out Examples that used a credit card
+    if b'Credit Card' in example['payment_type']:
+        return False
+    return True
+...
+
+filtered_examples = ExampleFilter(
+    examples=examples.output,
+    schema=schema.output,
+    module_file=filter_module,
```

Comment:
Is this the same assumption as Trainer (expects a function from a module path).

Reply:
Yes, it expects a `predicate_fn` (see the function defined just above this as an example)